platformio.ini

```diff
@@ -383,6 +381,14 @@ build_unflags = ${common.build_unflags}
 build_flags = ${common.build_flags} ${esp8266.build_flags} -D WLED_RELEASE_NAME=\"ESP02\"
 lib_deps = ${esp8266.lib_deps}
 
+[env:esp8266_2m_tasmota]
+board = esp_wroom_02
+platform = https://github.com/tasmota/platform-espressif8266#2024.09.00 ;; TODO: not sure this is the correct version
```

Comment:
What is the correct version we need for 8266?

Were we really using Arduino 3.1.2 for 8266 while only 2.0.9 for esp32?!

Reply:
Yes. IIRC @softhack007 had issues with newer Arduino cores.

Reply:
Confirmed - a few points are important:
* arduino for 8266 and arduino for esp32 are completely different - you can't compare their versions
* esp32 arduino 2.0.9 is "well proven" in use, while newer versions normally have new bugs. I've experimentally use 2.0.14 for some time and it seems to work, too. Beyond 2.0.14, there are new bugs again, like half-working serial, and some methods of the ESP class won't even compile for some boards.
* esp32 arduino v 3.x.y won't work for WLED. Too many API changes.

So esp32 arduino 2.0.14 looks most promising, and could be the right option for 0.16 development.

Arduino-esp32 v 3.0.x would be needed for supporting -C6. But in that case, I would suggest we switch to `pioarduino` which also comes from @Jason2866 and a few  others.

Reply:
PS2: arduino-esp32 v2.0.10 thru v2.0.13 are broken - I have stopped looking at them after experiencing some problems with audioreactive.

Reply:
Reverted back to 2.0.9 for esp32

Reply:
@Jason2866  If we want to stick with the 3.1.1 version of the Arduino Core for 8266 - is this possible with Tasmota?

Reply:
Don't get the question. Arduino core for eap8266 and esp32x are two complete different things. So yes, for esp8266 everything you want can be used as platform/framework.

Reply:
Unlike the release page for the esp32 builds, it's not so clear exactly what is in each release.

If the versions listed follow the Arduino Core version, then the newest is 2.7.8 and we ideally want to stay on 3.1.1 as that was what we were using before and all the code is expecting

Reply:
Basically we were using this release https://github.com/platformio/platform-espressif8266/releases/tag/v4.1.0 and we want to stay with the same 3.1.1 of Arduino - but I am unsure if any of the release for tasmota's platform-espressif8266 match this

Reply:
No Tasmota uses a modified Arduino esp8266 core 2.7.x version.
All 3.x versions uses more resources and Tasmota does not need features introduced with core 3.x. for esp8266. Furthermore we had stability issues with core 3.0.x when we tested with Tasmota. Probably this problems are fixed, but still there is no reason for Tasmota to use.